src/main/java/duke/Tasklist.java

```diff
@@ -0,0 +1,69 @@
+package duke;
+
+import duke.exceptions.NoSuchTaskException;
+import duke.tasks.DeadlinesTask;
+import duke.tasks.EventsTask;
+import duke.tasks.Task;
+import duke.tasks.ToDosTask;
+
+import java.util.ArrayList;
+public class Tasklist {
+    public ArrayList<Task> tasks;
+
+    public Tasklist() {
+        this.tasks = new ArrayList<>();
+    }
+
+    public void addTask(Task task){
+        tasks.add(task);
+    }
+
+    public void deleteTask(int index) throws NoSuchTaskException {
+        try {
+            tasks.remove(index-1);
+        } catch (IndexOutOfBoundsException e) {
+            throw new NoSuchTaskException("Task index is out of bounds");
+        }
+    }
+
+    public ArrayList<Task> getTasklist(){
+        return tasks;
+    }
+
+    public void doneTask(int index) throws NoSuchTaskException {
+        try {
+            Task temp = tasks.get(index-1);
+            temp.markDone();
+        } catch (IndexOutOfBoundsException e) {
+            throw new NoSuchTaskException("Task index is out of bounds");
+        }
+    }
+
+    public Integer getTasklistSize(){
+        return tasks.size();
+    }
+
+    public Task getTask(int index){
+        return tasks.get(index);
+    }
+
+
+
```

Comment:
Similar problems as above, indentation between methods and javadoc need to be improved.